### PR TITLE
helm: fix csiAddons enabled flag

### PR DESCRIPTION
**Pull request by peng225**
**Description of your changes:**

We should use csiAddons.enabled to check whether csiAddons is used.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (Madhu-1) [inline pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go:1]: this is not related to above change

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go (hunk 1 above)
Comment (peng225) [reply pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go]: Sorry, I deleted the line.

File: deploy/charts/rook-ceph/templates/role.yaml [hunk 2]
```diff
@@ -61,7 +61,7 @@ rules:
     resources: ["leases"]
     verbs: ["get", "watch", "list", "delete", "update", "create"]
 ---
-{{- if .Values.csi.csiAddons }}
+{{- if .Values.csi.csiAddons.enabled }}
```
[the remark this answers is not in the document]
Comment (peng225) [reply deploy/charts/rook-ceph/templates/role.yaml]: I fixed it using `and` operator instead of nested `if` blocks for readability.

File: deploy/charts/rook-ceph/templates/configmap.yaml [hunk 3]
```diff
@@ -76,7 +76,7 @@ data:
   CSI_VOLUME_REPLICATION_IMAGE: {{ .Values.csi.volumeReplication.image | quote }}
 {{- end }}
 {{- end }}
-{{- if .Values.csi.csiAddons }}
+{{- if .Values.csi.csiAddons.enabled }}
```
[the remark this answers is not in the document]
Comment (peng225) [reply deploy/charts/rook-ceph/templates/configmap.yaml]: Thank you for pointing it out. I reverted the line to the original code.

File: deploy/examples/common.yaml [hunk 4]
```diff
@@ -863,16 +863,6 @@ rules:
 ---
 kind: Role
 apiVersion: rbac.authorization.k8s.io/v1
-metadata:
```
Comment (Madhu-1) [inline deploy/examples/common.yaml:866]: these changes should not be removed or else the user needs to replay these permissions if he wants to enable csiaddons, in other words, no harm in keeping these by default.

Comment (peng225) [reply deploy/examples/common.yaml:866]: Though I only changed the condition of `if` statements at first, the CI complained that I should execute `make rbac` command and then push the generated changes. Code changes in this file were all generated by `make rbac` command.
Because `csiAddons.enabled` is set to `false` by default, I thought these changes are valid.
If my understanding is wrong, I would appreciate it if you could point it out.

Comment (Madhu-1) [reply deploy/examples/common.yaml:866]: @BlaineEXE @travisn @leseb  this could be a problem for the user to enable a few functionalities


* We need to update the documents to create the RBAC and enable the required sidecars
* OR create the RBAC rules by default only configmap change is fine to enable the functionality?

Comment (travisn) [reply deploy/examples/common.yaml:866]: It makes sense that these permissions should only be required if csi addons is enabled. If it is enabled, then they would need to update the helm chart anyway to update the RBAC. So it sounds like the issue is that the `make rbac` needs to enable this flag when it generates the common.yaml. @BlaineEXE Can we enable this flag for `make rbac`?

Comment (peng225) [reply deploy/examples/common.yaml:866]: @Madhu-1 @travisn @BlaineEXE Thank you for discussing this topic.
> @BlaineEXE Can we enable this flag for make rbac?

Do you mean to change the default value of csi.csiAddons.enabled flag to `true` in `values.yaml` file? If so, I agree with this idea.

@BlaineEXE I would be happy if you could comment on it.

Comment (travisn) [reply deploy/examples/common.yaml:866]: CSI addons should remain disabled by default since it requires independent CRDs to be installed. So we will need to find a way to keep the rbac optional in the helm chart based on that setting, but just add it to common.yaml always since we don't have a good way to make rbac more optional in the non-helm case.

Comment (peng225) [reply deploy/examples/common.yaml]: @travisn @BlaineEXE 
> Just add --set csiAddons.enabled, and you should be good to go.

This worked perfectly! `common.yaml` was reverted to the original state as expected.
I pushed the changes.

Two small things:
- The correct parameter was `csi.csiAddons.enabled`. Setting `csiAddons.enabled=true` did not worked well.
- The correct command to generate `common.yaml` was `makge gen-rbac`, not `make rbac`. Sorry for my mistake.

**Review by travisn — APPROVED** [review]

**Review by BlaineEXE — APPROVED** [review]

**Comment by peng225** [comment]
@Madhu-1 Could you check my code changes?

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
> @Madhu-1 Could you check my code changes?

Sorry I forgot about it.

**Comment by Madhu-1** [comment]
@travisn this is good to merge
